security-privacy-self-review.md

```diff
@@ -0,0 +1,77 @@
+
+
+Responses to the [Self-Review Questionnaire: Security and Privacy](https://w3ctag.github.io/security-questionnaire/) for the [Digital Goods API](https://github.com/WICG/digital-goods/)
+
+
+## 2.1 What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+The API allows an origin to query the details of digital goods available from a store backend, for that origin and user.
+It exposes the user's chosen currency (as part of the price details) and which items have been purchased.
+This information is needed for the origin to display purchases to the user and give the user access to features/purchases based on purchases made.
+
+The API also allows an origin to know that a purchase has been acknowledged, which is necessary to confirm that a transaction has completed.
+
+## 2.2 Is this specification exposing the minimum amount of information necessary to power the feature?
+
+Yes.
+
+## 2.3 How does this specification deal with personal information or personally-identifiable information or information derived thereof?
+
+The API involves an already-authenticated user and their purchases from that origin, but does expose any other information about the user. No information is persisted.
+
+## 2.4 How does this specification deal with sensitive information?
+
+No special treatment.
+
+## 2.5 Does this specification introduce new state for an origin that persists across browsing sessions?
+
+No? TODO Glen: double check this. How does auth work on the backend?
+
+## 2.6 What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+The origin will be able to make deductions from the presence or absence of the API.
+For example, a store backend may be available in certain contexts only, or a user agent may support the API on specific platforms only.
+
+## 2.7 Does this specification allow an origin access to sensors on a user’s device
+
+No.
+
+## 2.8 What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+Data about the user's digital goods purchases. If those purchases were made through the PaymentRequest API on the origin then it would have access to that information already.
+
+## 2.9 Does this specification enable new script execution/loading mechanisms?
+
+No.
+
+## 2.10 Does this specification allow an origin to access other devices?
+
+No.
+
+## 2.11 Does this specification allow an origin some measure of control over a user agent’s native UI?
+
+No.
+
+## 2.12 What temporary identifiers might this this specification create or expose to the web?
+
+None.
+
+## 2.13 How does this specification distinguish between behavior in first-party and third-party contexts?
+
+Supporting the API in third-party contexts is probably not necessary?
```

Comment:
From the [payment-request spec's "payment relevant browsing context"](https://w3c.github.io/payment-request/#dfn-payment-relevant-browsing-context) it looks like secure, top-level browsing contexts. But it also [mentions](https://w3c.github.io/payment-request/#cross-origin-payment-requests) delegating to a payment service through a cross-origin iframe. Wouldn't that be a non-top-level context? I'm confused.

I guess being top-level is more relevant to Payment Request, because it can request a browser UI. But happy to match what context it is using because this API is mostly useful in conjunction with that.

Reply:
It looks like Payment Request is activated by the permissions policy `"payment"`.

https://w3c.github.io/payment-request/#permissions-policy

That means it's available in top-level browsing contexts, _or_ any context that is delegated the `"payment"` permission from a top-level browsing context (transitively). Most new APIs work this way.

I think it's reasonable to make the same delegation here. We should just use the same permission policy, since we expect both to be used together. `"payment"`.

Reply:
Done.